database/security-rules.json

```diff
@@ -6,7 +6,8 @@
       "publicIds": {
         "$publicId": {
           ".validate": "newData.isString() && newData.val().matches(/^[-:a-zA-Z0-9]+$/) && $publicId.matches(/^[-a-z0-9]{3,20}$/) && ($publicId == null || newData.parent().parent().parent().child('auth').child('publicIds').child($publicId).val() == newData.val() && newData.parent().parent().parent().child('auth').child('usedPublicIds').child($publicId).val() == true && newData.parent().parent().parent().child('auth').child('users').child(newData.val()).child('publicId').val() == $publicId)",
-          ".write": "auth != null && auth.uid == newData.val() && (newData.val() != null && data.val() == null)"
+          ".write": "auth != null && auth.uid == newData.val() && (newData.val() != null && data.val() == null)",
+          ".read": "auth != null"
```

Comment:
I am not sure we should.

It's fine for new database; I think the new UID format is unique to each DB and I am assuming it's opaque. For firebase DB imported from the legacy Firebase console, old UID are not unique and should be consider private data (you can track user from different DB). It probably expose the google user UID.